topic-operator/src/main/java/io/strimzi/operator/topic/cruisecontrol/CruiseControlClient.java

```diff
@@ -0,0 +1,158 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.operator.topic.cruisecontrol;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import io.strimzi.api.kafka.model.topic.KafkaTopic;
+
+import java.net.http.HttpResponse;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+/**
+ * Cruise Control REST API client.
+ * <br/><br/>
+ * The server runs one task execution at a time, additional 
+ * requests are queued up to {@code max.active.user.tasks}.
+ */
+public interface CruiseControlClient extends AutoCloseable {
+    /**
+     * HTTP request timeout in seconds.
+     */
+    long HTTP_REQUEST_TIMEOUT_SEC = 60;
+
+    /**
+     * Send a POST request to {@code topic_configuration} endpoint.
+     * This can be used to request replication factor changes.
+     * 
+     * @param kafkaTopics List of Kafka topics.
+     * @return The user task id.
+     */
+    String topicConfiguration(List<KafkaTopic> kafkaTopics);
+
+    /**
+     * Send a GET request to {@code user_tasks} endpoint.
+     * This can be used to check the asynchronous task execution result.
+     * 
+     * @param userTaskIds List of user task ids.
+     * @return User tasks response.
+     */
+    UserTasksResponse userTasks(List<String> userTaskIds);
+
+    /**
+     * Get the error message from HTTP response.
+     * 
+     * @param response The HTTP response.
+     * @return The error message.
+     */
+    Optional<String> errorMessage(HttpResponse<String> response);
+    
+    /**
+     * Topic names grouped by replication factor value.
+     * 
+     * @param topicByReplicationFactor Topic names grouped by replication factor value.
+     */
+    record ReplicationFactor(@JsonProperty("topic_by_replication_factor") Map<Integer, String> topicByReplicationFactor) { }
```

Comment:
`Map<Integer, String>` but the the value is some topic names (plural)? How does that work? Are the names separated by some character (if so, you should define it!). Why not `Map<Integer, Collection<String>>`?

Reply:
> Yes. In order to support batch requests, we send a JSON payload where, for each RF value, we have a simple regex like `topic1|topic2|topic3` (i.e. a group by operation).

This needs to be in the javadoc.